cpu/rp2350/include/clock_conf.h

```diff
@@ -0,0 +1,157 @@
+/** Post divider power down bits for the PLL */
+#define PLL_PWR_POSTDIVPD_BITS 0x00000008u
+/** Enable bit for the peripheral clock control register */
+#define CLK_PERI_CTRL_ENABLE_BIT (1u << 11u)
+/** Default CPU frequency in Hz, set to 125 MHz as per hardware manual */
+#define CPUFREQ 125000000u
+/** Maximum crystal frequency */
+#define CLOCK_XOSC_MAX MHZ(15u)
+/** Minimum crystal frequency */
+#define CLOCK_XOSC_MIN MHZ(5u)
+/** Crystal frequency */
+#define CLOCK_XOSC (XOSC_HZ)
+/** Minimum value of the post PLL clock divers */
+#define PLL_POSTDIV_MIN 1u
+/** Maximum value of the post PLL clock divers */
+#define PLL_POSTDIV_MAX 7u
+/** Minimum value of the PLL VCO feedback scaler */
+#define PLL_VCO_FEEDBACK_SCALE_MIN 16u
+/** Maximum value of the PLL VCO feedback scaler */
+#define PLL_VCO_FEEDBACK_SCALE_MAX 320u
+/** Minimum value of the clock divider applied before
+ * feeding in the reference clock into the PLL */
+#define PLL_REF_DIV_MIN 1u
+/** Minimum value of the clock divider applied before feeding in
+ *  the reference clock into the PLL */
+#define PLL_REF_DIV_MAX 1u
+/** PLL feedback divider value, set to 125 as per hardware manual */
+#define PLL_FEEDBACK_DIVIDER_VALUE 125u
+/** Enable bit for the system clock control register to select the peripheral
+ * clock */
+#define CLK_SYS_PERI_CTRL_ENABLE_BIT (1u << 0u)
+/** Selected field value for the system clock control register
+ * to select the peripheral clock */
+#define CLK_SYS_SELECTED_PERI_FIELD_VALUE 2u
+/** RIOT core clock frequency defined as the CPU frequency */
+#define CLOCK_CORECLOCK MHZ(12u)
+
+#if (PLL_VCO_FEEDBACK_SCALE_MIN < PLL_VCO_FEEDBACK_SCALE_MIN) || \
+    (PLL_VCO_FEEDBACK_SCALE_MAX > PLL_VCO_FEEDBACK_SCALE_MAX)
+#  error "Value for PLL_VCO_FEEDBACK_SCALE out of range, check config"
+#endif
+#if (PLL_REF_DIV_MIN < PLL_REF_DIV_MIN) || (PLL_REF_DIV_MAX > PLL_REF_DIV_MAX)
+#  error "Value for PLLxosc_sleep_REF_DIV out of range, check config"
+#endif
+#if (PLL_POSTDIV_MIN < PLL_POSTDIV_MIN) || (PLL_POSTDIV_MAX > PLL_POSTDIV_MAX)
+#  error "Value for PLL_POSTDIV out of range, check config"
+#endif
+#if ((CLOCK_XOSC > CLOCK_XOSC_MAX) || (CLOCK_XOSC < CLOCK_XOSC_MIN))
+#  error "Value for CLOCK_XOSC out of range, check config"
+#endif
+
+/** Post divider for the PLL, calculated based on the post divider values */
+#define PDIV ((PLL_PD1 << PLL_PRIM_POSTDIV1_LSB) | (PLL_PD2 << PLL_PRIM_POSTDIV2_LSB))
+/** Feedback divider for the PLL, calculated based on the VCO frequency and
+ * reference clock frequency */
+#define FBDIV ((PLL_VCO_FREQ / XOSC_HZ) / PLL_REF_DIV)
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Configures the Crystal to run.
+ * @see The reference hardware manual suggests to use a 12 MHz crystal, which we
+ * use by default.
+ */
+void xosc_start(void);
+
+/**
+ * @brief   Stop the crystal.
+ */
+void xosc_stop(void);
+
+/**
+ * @brief   Sleep for a given number of cycles.
+ */
+void xosc_sleep(int32_t cycles);
```

Comment:
```suggestion
/**
 * @brief   Sleep for a given time in milliseconds.
 */
void xosc_sleep(uint32_t milliseconds);
```

It makes sense to use an `uint32_t` here, because you can't sleep for negative time anyways.